contracts/strategies/StrategyBase.sol

```diff
@@ -0,0 +1,231 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+pragma experimental ABIEncoderV2;
+
+// Hardhat
+import "hardhat/console.sol";
+
+// Lyra
+import {VaultAdapter} from "@lyrafinance/core/contracts/periphery/VaultAdapter.sol";
+import {GWAVOracle} from "@lyrafinance/core/contracts/periphery/GWAVOracle.sol";
+
+// Libraries
+import {Vault} from "../libraries/Vault.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {LyraVault} from "../core/LyraVault.sol";
+import {DecimalMath} from "@lyrafinance/core/contracts/synthetix/DecimalMath.sol";
+import {SignedDecimalMath} from "@lyrafinance/core/contracts/synthetix/SignedDecimalMath.sol";
+
+contract StrategyBase is VaultAdapter {
+  using DecimalMath for uint;
+  using SignedDecimalMath for int;
+
+  LyraVault public immutable vault;
+  OptionType public immutable optionType;
+  GWAVOracle public immutable gwavOracle;
+
+  /// @dev asset used as collateral in AMM to sell. Should be the same as vault asset
+  IERC20 public collateralAsset;
+
+  mapping(uint => uint) public lastTradeTimestamp;
+
+  uint public activeExpiry;
+  uint[] public activeStrikeIds;
+  mapping(uint => uint) public strikeToPositionId;
+
+  ///////////
+  // ADMIN //
+  ///////////
+
+  modifier onlyVault() virtual {
+    require(msg.sender == address(vault), "only Vault");
+    _;
+  }
+
+  constructor(
+    LyraVault _vault,
+    OptionType _optionType,
+    GWAVOracle _gwavOracle
+  ) VaultAdapter() {
+    vault = _vault;
+    optionType = _optionType;
+    gwavOracle = _gwavOracle;
+  }
+
+  function initAdapter(
+    address _curveSwap,
+    address _optionToken,
+    address _optionMarket,
+    address _liquidityPool,
+    address _shortCollateral,
+    address _synthetixAdapter,
+    address _optionPricer,
+    address _greekCache,
+    address _quoteAsset,
+    address _baseAsset,
+    address _feeCounter
+  ) external onlyOwner {
+    // set addressese for LyraVaultAdapter
+    setLyraAddresses(
+      _curveSwap,
+      _optionToken,
+      _optionMarket,
+      _liquidityPool,
+      _shortCollateral,
+      _synthetixAdapter,
+      _optionPricer,
+      _greekCache,
+      _quoteAsset,
+      _baseAsset,
+      _feeCounter
+    );
+
+    quoteAsset.approve(address(vault), type(uint).max);
+    baseAsset.approve(address(vault), type(uint).max);
+    collateralAsset = _isBaseCollat() ? baseAsset : quoteAsset;
+  }
+
+  ///////////////////
+  // VAULT ACTIONS //
+  ///////////////////
+
+  /**
+   * @dev set the expiry for the next round
+   */
+  function _setExpiry(uint expiry) internal {
```

Comment:
to be honest, I think this is too specific too. Thinking of vaults that may trade several expiries

Reply:
agree and updated